### PR TITLE
chore: change default factor to compute memory size

### PR DESCRIPTION
**Pull request by shuiyisong**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

When the specifications of the host machine or docker container are relatively small, for example 1C2G, the default `global_write_buffer_size` would be computed to 32MB, which is too small for a database. This pr mainly shrink the factor so that the default size is 8 times larger. In a 1C2G case it would now be 256MB of `global_write_buffer_size`.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/mito2/src/config.rs [hunk 1]
```diff
@@ -193,13 +193,13 @@ impl MitoConfig {
 
     fn adjust_buffer_and_cache_size(&mut self, sys_memory: ReadableSize) {
         // Use 1/64 of OS memory as global write buffer size, it shouldn't be greater than 1G in default mode.
```
Comment (v0y4g3r) [inline src/mito2/src/config.rs:195]: Also change the doc here: 
```suggestion
        // Use 1/8 of OS memory as global write buffer size, it shouldn't be greater than 1G in default mode.
```

File: src/mito2/src/config.rs [hunk 2]
```diff
@@ -193,13 +193,13 @@ impl MitoConfig {
...
-        let global_write_buffer_size = cmp::min(sys_memory / 64, ReadableSize::gb(1));
+        let global_write_buffer_size = cmp::min(sys_memory / 8, ReadableSize::gb(1));
         // Use 2x of global write buffer size as global write buffer reject size.
         let global_write_buffer_reject_size = global_write_buffer_size * 2;
         // Use 1/256 of OS memory size as SST meta cache size, it shouldn't be greater than 128MB in default mode.
```
Comment (v0y4g3r) [inline src/mito2/src/config.rs:199]:
```suggestion
        // Use 1/32 of OS memory size as SST meta cache size, it shouldn't be greater than 128MB in default mode.
```

File: src/mito2/src/config.rs [hunk 3]
```diff
@@ -193,13 +193,13 @@ impl MitoConfig {
...
-        let sst_meta_cache_size = cmp::min(sys_memory / 256, ReadableSize::mb(128));
+        let sst_meta_cache_size = cmp::min(sys_memory / 32, ReadableSize::mb(128));
         // Use 1/128 of OS memory size as mem cache size, it shouldn't be greater than 512MB in default mode.
```
Comment (v0y4g3r) [inline src/mito2/src/config.rs:201]:
```suggestion
        // Use 1/16 of OS memory size as mem cache size, it shouldn't be greater than 512MB in default mode.
```

File: src/mito2/src/config.rs (hunk 1 above)
Comment (killme2008) [reply src/mito2/src/config.rs]: I think it's better to remove these comments and add some constants for these factors.

Comment (killme2008) [reply src/mito2/src/config.rs]: Move the comments to the constants.

**Comment by evenyag** [comment]
Do we need to update the example config files?

https://github.com/GreptimeTeam/greptimedb/blob/278e4c8c3090358341b82b6963b7877e86266793/config/datanode.example.toml#L97-L106

Also see https://github.com/GreptimeTeam/greptimedb/blob/main/config/standalone.example.toml

**Review by QuenKar — APPROVED** [review]
LGTM

**Comment by shuiyisong** [comment]
> Do we need to update the example config files?
> 
> https://github.com/GreptimeTeam/greptimedb/blob/278e4c8c3090358341b82b6963b7877e86266793/config/datanode.example.toml#L97-L106
> 
> Also see https://github.com/GreptimeTeam/greptimedb/blob/main/config/standalone.example.toml

`# Global write buffer size for all regions. If not set, it's default to 1/8 of OS memory.`

**Comment by QuenKar** [comment]
> > Do we need to update the example config files?
> > https://github.com/GreptimeTeam/greptimedb/blob/278e4c8c3090358341b82b6963b7877e86266793/config/datanode.example.toml#L97-L106
> > 
> > Also see https://github.com/GreptimeTeam/greptimedb/blob/main/config/standalone.example.toml
> 
> `# Global write buffer size for all regions. If not set, it's default to 1/8 of OS memory.` Is this ok?

it also has a max limit `1GB` in default.

**Comment by shuiyisong** [comment]
@v0y4g3r @evenyag @killme2008 PTAL _(:з」∠)_

File: src/mito2/src/config.rs [hunk 4]
```diff
@@ -33,6 +33,13 @@ const MULTIPART_UPLOAD_MINIMUM_SIZE: ReadableSize = ReadableSize::mb(5);
 /// Default channel size for parallel scan task.
 const DEFAULT_SCAN_CHANNEL_SIZE: usize = 32;
 
+// Use 1/8 of OS memory as global write buffer size in default mode
```
Comment (killme2008) [inline src/mito2/src/config.rs:36]:
```suggestion
// Use `1/GLOBAL_WRITE_BUFFER_SIZE_FACTOR` of OS memory as global write buffer size in default mode
```

File: src/mito2/src/config.rs [hunk 5]
```diff
@@ -33,6 +33,13 @@ const MULTIPART_UPLOAD_MINIMUM_SIZE: ReadableSize = ReadableSize::mb(5);
...
+const GLOBAL_WRITE_BUFFER_SIZE_FACTOR: u64 = 8;
+/// Use 1/32 of OS memory size as SST meta cache size in default mode
```
Comment (killme2008) [inline src/mito2/src/config.rs:38]:
```suggestion
/// Use `1/SST_META_CACHE_SIZE_FACTOR` of OS memory size as SST meta cache size in default mode
```

File: src/mito2/src/config.rs [hunk 6]
```diff
@@ -33,6 +33,13 @@ const MULTIPART_UPLOAD_MINIMUM_SIZE: ReadableSize = ReadableSize::mb(5);
...
+const SST_META_CACHE_SIZE_FACTOR: u64 = 32;
+/// Use 1/16 of OS memory size as mem cache size in default mode
```
Comment (killme2008) [inline src/mito2/src/config.rs:40]:
```suggestion
/// Use `1/MEM_CACHE_SIZE_FACTOR` of OS memory size as mem cache size in default mode
```

**Review by killme2008 — APPROVED** [review]
Almost LGTM

**Pull request merged**